app/Http/Middleware/CheckRole.php

```diff
@@ -0,0 +1,24 @@
+<?php
+
+namespace App\Http\Middleware;
+
+use Closure;
+use Illuminate\Support\Facades\Auth;
+
+class CheckRole
+{
+    /**
+     * Handle an incoming request.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @param  \Closure  $next
+     * @return mixed
+     */
+    public function handle($request, Closure $next, $role)
+    {
+        if (Auth::user()->role == $role) {
+            return redirect()->route('chat.user');
```

Comment:
šitą middleware turbūt naudosim ne tik su chatais, tai reikėtų, kad labiau general purpose būtų. pvz `abort(403)`